### PR TITLE
[cxx-interop] NFC: Add doc comments for overlay protocols

### PR DESCRIPTION
**Pull request by egorzhdan**

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

File: stdlib/public/Cxx/CxxDictionary.swift [hunk 1]
```diff
@@ -10,6 +10,10 @@
 //
 //===----------------------------------------------------------------------===//
 
+/// A C++ type that represents a dictionary.
```
Comment (hyp) [inline stdlib/public/Cxx/CxxDictionary.swift:13]: it seems strange to say "A C++ type..." when this is defining a protocol.

Comment (egorzhdan) [reply stdlib/public/Cxx/CxxDictionary.swift:13]: This seems common in the stdlib, for instance:

```
/// A type that provides sequential, iterated access to its elements.
/// ...
public protocol Sequence<Element> {
```

```
/// A type that can be initialized with an integer literal.
/// ...
public protocol ExpressibleByIntegerLiteral {
```

```
/// A type with values that support addition and subtraction.
/// ...
public protocol AdditiveArithmetic: Equatable {
```

Comment (hyp) [reply stdlib/public/Cxx/CxxDictionary.swift:13]: I see, that sounds good.

**Review by hyp — APPROVED** [review]

**Pull request merged**